flink-test-utils-parent/flink-test-utils-junit/src/test/java/org/apache/flink/core/testutils/CommonTestUtilsTest.java

```diff
@@ -45,7 +47,29 @@ public void testObjectFromNewClassLoaderObject() throws Exception {
 
 	@Test
 	public void testObjectFromNewClassLoaderClassLoaders() throws Exception {
-		final CommonTestUtils.ObjectAndClassLoader objectAndClassLoader = CommonTestUtils.createObjectFromNewClassLoader();
+		final CommonTestUtils.ObjectAndClassLoader<Serializable> objectAndClassLoader = CommonTestUtils.createSerializableObjectFromNewClassLoader();
+
+		assertNotEquals(ClassLoader.getSystemClassLoader(), objectAndClassLoader.getClassLoader());
+		assertEquals(ClassLoader.getSystemClassLoader(), objectAndClassLoader.getClassLoader().getParent());
+	}
+
+	@Test
+	public void testExceptionObjectFromNewClassLoaderObject() throws Exception {
+		final CommonTestUtils.ObjectAndClassLoader<Exception> objectAndClassLoader = CommonTestUtils.createExceptionObjectFromNewClassLoader();
+		final Object o = objectAndClassLoader.getObject();
+
+		assertNotEquals(ClassLoader.getSystemClassLoader(), o.getClass().getClassLoader());
+
+		try {
+			Class.forName(o.getClass().getName());
+			fail("should not be able to load class from the system class loader");
```

Comment:
could we just have this check in other tests/one helper test method?

Reply:
it's actually quite useful as a separate test, since you see at a glance whether the core assumption holds or not.

Reply:
ok, I see, it is actually about object and class loader test separation.